### DISCUSSION TITLE
Allow domoticz mqtt autodiscovery

### DISCUSSION THREAD

**Discussion by rezzalex** [tsightler/ring-mqtt#873] [Feature Requests]
Hello, 

Do you think you could implement the domoticz mqtt autodiscovery ?

I tried to listen to the "will" discovery topic from domoticz and it does not allow to autodiscover anything.
Basically, the topic contains only one "Status" field. Is it normal ?

It seems ha autodiscovery and DZ autodiscovery should be the same .. what did I miss?

please see the same thing working for EzViz Cam, (a lot of topics in the discovery message and almost nothing for RING)

![image](https://github.com/user-attachments/assets/46dc9d54-01c4-4177-b49a-bcb20068a4a9)

Thanks

**Comment by tsightler** [comment]
Better to ask domoticz questions on their forum vs here as I know nothing about it.  My understanding is that it support Home Assistant style MQTT discovery and you just have to configure an MQTT discovery gateway which is configured for the discovery topic (ring-mqtt uses the Home Assistant default discovery topic which is `homeassistant`.

I'm not really sure why you are posting images of some other device, I don't care about it.  You can clearly see that homeassistant and ring top level topics both have multiple sub-topics, so open those if you want to see state of Ring devices.

**Reply by rezzalex** [reply]
Thank you @tsightler  for the quick answer.

The image is for clearly seeing that the other discovery topic (dz_ezviz) contains a lot of informations used successfully by DZ mqtt autodiscovery plugin.
Compared to the ring autodiscovery topic (ylthat I have named "dz_ring")  contains barely nothing (just the "status online")

**Reply by tsightler** [reply]
As I already stated, dz_ring is not the discovery topic, the discovery topic is homeassistant, and you can tell from your screenshot that it has plenty of topics and messages.  I have no idea what dz_ring is, but I do not believe it is from ring-mqtt.

> ...that I have named "dz_ring"...

How exactly did you do this as ring-mqtt has no setting to change the discovery topic from the default of homeassistant?  Did you modify the code directly?

**Reply by rezzalex** [reply]
I am using the docker version and the discovery topic seems to be configurable, so I have changed it so that DZ can distinguish the different origins of mqtt devices :  zwavejs, ez-viz, and ring

**Comment by rezzalex** [comment]
DZ autodiscovery works well with the default parameter : 

"hass_topic": "homeassistant/status",

As a consequence, in DZ, because it has the same discovery topic as ZwaveJsUI, bot ZwaveJsUI and Ring DZ devices are regrouped under the same "dz hardware" (that I originally named "zwave").
So it seems your ring mqtt autodiscovery does not work if try to change the auto discovers topic, but as other projects allow it, I could use their features to separate the devices.
Or while changing the discovery topic, I could have missed a pattern that needs to be respected ?
It is not so Important anyway, just a matter of internal devices organization.

Thanks for your work
